spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/MessageChannelToInputObservableArgumentAdapter.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.reactive;
+
+import reactor.adapter.RxJava1Adapter;
+import rx.Observable;
+
+import org.springframework.cloud.stream.annotation.Input;
+import org.springframework.cloud.stream.binding.StreamListenerArgumentAdapter;
+import org.springframework.core.MethodParameter;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.SubscribableChannel;
+
+/**
+ * Adapts an {@link org.springframework.cloud.stream.annotation.Input} annotated
+ * {@link MessageChannel} to an {@link Observable}.
```

Comment:
could mention here that it does so by wrapping the Flux adapter and using Reactor's own adapter support